src/main/java/Duke.java

```diff
@@ -1,10 +1,164 @@
+                " \\      \\   ___________|__|____  |  | _|__|\n" +
+                " /   |   \\ /  _ \\_  __ \\  \\__  \\ |  |/ /  |\n" +
+                "/    |    (  <_> )  | \\/  |/ __ \\|    <|  |\n" +
+                "\\____|__  /\\____/|__|  |__(____  /__|_ \\__|\n" +
+                "        \\/                     \\/     \\/   \n";
+
         System.out.println("Hello from\n" + logo);
+        printMessage(greetMessage);
+    }
+
+    /**
+     * Prints goodbye message.
+     */
+    public static void goodbye(){
+        String goodbyeMessage = "Bye! Hope to see you again soon! MEGANE!!";
+
+        printMessage(goodbyeMessage);
+    }
+
+    /**
+     * Reads user input.
+     *
+     * @return String User input.
+     */
+    public static String readInput(){
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+        return line;
+    }
+
+    /**
+     * Reads user input and prints user input.
+     */
+    public static void echo(){
+        boolean isRunning = true;
+        while(isRunning){
+            String input = readInput();
+            if(input.equalsIgnoreCase("bye")) {
+                isRunning = false;
+            }
+            else printMessage(input);
+        }
+    }
+
+    /**
+     * Prints list of tasks.
+     *
+     * @param taskList List of tasks to be printed.
+     */
+    public static void printList(List<Task> taskList){
+        printLine();
+        for(int i = 0; i < taskList.size(); i++){
+            Task task = taskList.get(i);
+            System.out.println((i + 1) + ". [" + task.getStatusIcon() + "] " + task.getDescription());
+        }
+        printLine();
+    }
+
+    /**
+     * Adds a task to list of tasks.
+     *
+     * @param taskList List of tasks to be printed.
+     * @param description description of task to be added.
+     */
+    public static void addToList(List<Task> taskList, String description){
+        Task newTask = new Task(description);
+        taskList.add(newTask);
+        printMessage("added: " + description);
+    }
+
+    /**
+     * Marks task as done or undone.
+     *
+     * @param taskList List of tasks.
+     * @param instruction User instruction on which task to mark.
+     */
+    public static void markTask(List<Task> taskList, String instruction) {
+        boolean done = true;
+        String message;
+
+        if (instruction.substring(0, 4).equalsIgnoreCase("mark")) {
+            done = true;
+        } else if (instruction.substring(0, 6).equalsIgnoreCase("unmark")) {
+            done = false;
+        }
+
+        int index = (new Scanner(instruction).useDelimiter("\\D+").nextInt()) - 1;
+        message = done ? "Nice! I've marked this task as done:\n" + "   [X] "
+                + taskList.get(index).getDescription() :
+                " OK, I've marked this task as not done yet:\n" + "   [ ] "
+                + taskList.get(index).getDescription();
+
+        try {
```

Comment:
For this catch exception it may be better to include the message part above inside the "try" section as well, since it could also lead to invalid index access and such.